File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/2. Interview Problems/python/02. Exceptionally odd.py
"""QUESTION:
Given an array where every number appears an even number of times except one, find that exceptional number.

Approach:
XOR all elements. Pairs cancel out, leaving the number with odd frequency.

Time Complexity: O(n)
Space Complexity: O(1)
"""
from typing import List

def getOddOccurrence(arr: List[int]) -> int:
    xr = 0
    for num in arr:
        xr ^= num
    return xr

File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/2. Interview Problems/javascript/02. Exceptionally odd.js
/**
QUESTION:
Given an array of positive integers where every value appears an even number of times except one, return that exceptional value.

Approach:
Bitwise XOR all elements. Pairs cancel out (x ^ x = 0), so the remaining value is the element with odd frequency.

Time Complexity: O(n)
Space Complexity: O(1)
*/

function getOddOccurrence(arr) {
    let xr = 0;
    for (const num of arr) {
        xr ^= num;
    }
    return xr;
}

module.exports = { getOddOccurrence };
